src/wp-admin/includes/class-wp-automatic-updater.php

```diff
@@ -486,6 +520,86 @@ public function update( $type, $item ) {
 			}
 		}
 
+		if ( 'theme' === $type ) {
+			error_log( '    Theme ' . var_export( $item->theme, true ) . ' has been upgraded.' );
+		}
+
+		if ( 'plugin' === $type ) {
+			error_log( '    Plugin ' . var_export( $item->slug, true ) . ' has been upgraded.' );
+			if ( is_plugin_inactive( $upgrader_item ) ) {
+				error_log( '    ' . var_export( $upgrader_item, true ) . ' is inactive and will not be checked for fatal errors.' );
+			}
+
+			if ( $was_active && ! is_wp_error( $upgrade_result ) ) {
+
+				/*
+				 * The usual time limit is five minutes. However, as a loopback request
+				 * is about to be performed, increase the time limit to account for this.
+				 */
+				if ( function_exists( 'set_time_limit' ) ) {
+					set_time_limit( 10 * MINUTE_IN_SECONDS );
+				}
+
+				// Avoid a race condition when there are 2 sequential plugins that have fatal errors.
+				sleep( 2 );
```

Comment:
Is there further information that can be added to the comment above this magic number?

Reply:
@afragen

Reply:
In previous testing it was possible that rollbacks overlapped. This pause is also used in `move_dir` due to an issue in VirtualBox. 

It came about when testing against multiple plugin fatals requiring rollback. 

https://github.com/WordPress/rollback-update-failure/commit/9ed09ea855cec8e3d733702c168bca6625412c1b